crates/ruff_python_parser/src/lexer.rs

```diff
@@ -555,6 +587,9 @@ impl<'src> Lexer<'src> {
         // SAFETY: Safe because the function is only called when `self.fstrings` is not empty.
         let fstring = self.fstrings.current().unwrap();
 
+        // Keep the current flags in sync throughout the f-string context.
+        self.current_flags = fstring.flags();
+
```

Comment:
This means that all three f-string tokens (`FStringStart`, `FStringMiddle`, `FStringEnd`) will have the flag set without any additional cost.